docs/encyclopedia/application-message-passing.mdx

```diff
@@ -26,57 +28,85 @@ keywords:
+### Sending Signals
 
-- A Workflow Id
-- Workflow input
-- A [Signal](#signals) name
-- Signal input
+You can send Signals from any Temporal Client, the Temporal CLI, or you can Signal one Workflow to another.
+You can also "Signal-With-Start" and send a Signal from
 
-If there is a running Workflow Execution with the given Workflow Id, it will be Signaled.
-Otherwise, a new Workflow Execution is started and immediately sent the Signal.
+#### Send a Signal from a Temporal Client or the CLI
+
+<RelatedReadContainer>
+    <RelatedReadItem path="/cli/workflow#signal" text="Send a Signal using the Temporal CLI" archetype="feature-guide" />
+    <RelatedReadItem path="/develop/go/message-passing#send-signal-from-client" text="Send Signals with the Go SDK" archetype="feature-guide" />
+    <RelatedReadItem path="/develop/java/message-passing#send-signal-from-client" text="Send Signals with the Java SDK" archetype="feature-guide" />
+    <RelatedReadItem path="/develop/php/message-passing#send-signal-from-client" text="Send Signals with the PHP SDK" archetype="feature-guide" />
+    <RelatedReadItem path="/develop/python/message-passing#send-signal-from-client" text="Send Signals with the Python SDK" archetype="feature-guide" />
+    <RelatedReadItem path="/develop/typescript/message-passing#send-signal-from-client" text="Send Signals with the TypeScript SDK" archetype="feature-guide" />
+    <RelatedReadItem path="/develop/dotnet/message-passing#send-signal-from-client" text="Send Signals with the .NET SDK" archetype="feature-guide" />
+</RelatedReadContainer>
+
+#### Send a Signal from one Workflow to another.
+
+<RelatedReadContainer>
+    <RelatedReadItem path="/develop/go/message-passing#send-signal-from-workflow" text="Send Signals from Workflows with the Go SDK" archetype="feature-guide" />
+    <RelatedReadItem path="/develop/java/message-passing#send-signal-from-workflow" text="Send Signals from Workflows with the Java SDK" archetype="feature-guide" />
+    <RelatedReadItem path="/develop/php/message-passing#send-signal-from-workflow" text="Send Signals from Workflows with the PHP SDK" archetype="feature-guide" />
+    <RelatedReadItem path="/develop/python/message-passing#send-signal-from-workflow" text="Send Signals from Workflows with the Python SDK" archetype="feature-guide" />
+    <RelatedReadItem path="/develop/typescript/message-passing#send-signal-from-workflow" text="Send Signals from Workflows with the TypeScript SDK" archetype="feature-guide" />
+    <RelatedReadItem path="/develop/dotnet/message-passing#send-signal-from-workflow" text="Send Signals from Workflows with the .NET SDK" archetype="feature-guide" />
+</RelatedReadContainer>
+
+#### Signal-With-Start
+
+Signal-With-Start is a great tool for lazily creating workflows. When you send this operation, if there is a running Workflow Execution with the given Workflow Id, it will be Signaled. Otherwise, a new Workflow Execution is started and immediately sent the Signal.
```

Comment:
To my mind, "lazily" isn't quite accurate here. In "lazy evaluation", there is no doubt that the evaluation will occur; it's just that it's deferred until needed. I'd prefer to say that Signal-With-Start is "get or create" semantics. So I'd change this to

```suggestion
When you send this operation, if there is a running Workflow Execution with the given Workflow Id, it will be Signaled. Otherwise, a new Workflow Execution is started and immediately sent the Signal.
```

Reply:
Per Claude: "In programming, the "get-or-create" pattern is commonly referred to as the "lazy initialization" pattern."
I'll use the more precise term.